telemetry/metrics.go

```diff
@@ -88,17 +106,39 @@ func New(cfg Config) (_ *Metrics, rerr error) {
 	metricsConf := metrics.DefaultConfig(cfg.ServiceName)
 	metricsConf.EnableHostname = cfg.EnableHostname
 	metricsConf.EnableHostnameLabel = cfg.EnableHostnameLabel
-
-	memSink := metrics.NewInmemSink(10*time.Second, time.Minute)
-	inMemSig := metrics.DefaultInmemSignal(memSink)
-	defer func() {
-		if rerr != nil {
-			inMemSig.Stop()
+	var sink metrics.MetricSink
+
+	switch cfg.MetricsBackend {
+	case "mem":
```

Comment:
Nit: can we put the types in const at the type of the file pls? Or even an enum.

```go
const (
  MetricsBackendMem 
)
```